src/app/map/Tile/Item/item.tsx

```diff
@@ -88,28 +90,56 @@ export const DynamicItemTile = ({
   const isDraggable = interactive && tileActions?.canDragTile(item.metadata.coordId) === true;
   const isBeingDragged = tileActions?.isDraggingTile(item.metadata.coordId) === true;
   
+  // Check if this tile is a valid drop target for swapping
```

Comment:
All the code in this DynamicItemTile relative to drag and drop should be in a helper function and applied in the render but should not occupy much space and save cognitive load